course-instructions.md

```diff
@@ -0,0 +1,760 @@
+
+### What's ahead in pipelines II
+
+In this training, the focus will be on conventions and best practices for making better, smarter pipelines for USGS Data Science projects. You'll learn new things here that will help you refine your knowledge from the first class and put it into practice. Let's get started!
+
+
+:keyboard: Activity: Add collaborators to get started.
+
+As with pipelines I, please invite a few collaborators to your repository so they can easily comment and review in the future. In the :gear: Settings widget at the top of your repo, select "Manage access". Go ahead and invite your course instructor. It should look something like this: 
+![add some friends](https://user-images.githubusercontent.com/2349007/81471981-c0094900-91ba-11ea-93b0-0ffd31ec4ea9.png)
+
+<hr>
+
+### Set up your local environment before continuing
+
+During the course, we will ask you to build the pipeline, explore how to troubleshoot, and implement some of the best practices you are learning. To do this, you will work with the pipeline locally and commit/push your changes to GitHub for review. 
+
+See details below for how to get started working with code and files that exist within the course repsository:
+
+---
+
+Open a git bash shell (Windows:diamond_shape_with_a_dot_inside:) or a terminal window (Mac:green_apple:) and change (`cd`) into the directory you work in for projects in R (for me, this is `~/Documents/R`). There, clone the repository and set your working directory to the new project folder that was created:
+
+```
+git clone git@github.com:<your-username-here>/ds-pipelines-targets-2-course-static.git
+cd ds-pipelines-targets-2
+```
+
+You can also open this project in RStudio by double-clicking the .Rproj file in the `ds-pipelines-targets-2` directory.
+
+:star: Now you have the repository locally! Follow along with the commands introduced and make changes to the code as requested throughout the remainder of the course.
+
+</details>
+
+<hr> 
+
+<details><summary><h2>How to get past the gotchas without getting gotten again</h2></summary>
+
+In this course, we're going to go one by one through a series of tips that will help you avoid common pitfalls in pipelines. These tips will help you in the next sections and in future work. A quick list of what's to come:
+
+- :mag: How to debug in a pipeline
+- :eyes: Visualizing and understanding the status of dependencies in a pipeline
+- :speech_balloon: `tar_visnetwork()` and `tar_outdated()` to further interrogate the status of pipeline targets
+- :arrows_clockwise: What is a cyclical dependency and how do I avoid it?
+- :warning: Undocumented file output from a function 
+- :open_file_folder: Using a directory as a dependency 
+- :clipboard: How do I know when to use an object vs a file target or even use a target at all? 
+- :gear: USGS Data Science naming conventions
+- :unlock: Final tips for smart pipelining
+
+<hr>
+
+### How to inspect parts of the pipeline and variables within functions
```

Comment:
Is there any content that would be worth adding to this after the mural brainstorm on debugging in DSP?

Reply:
I'll take a closer look with a bias toward "adding nothing" (thinking about info overload here) #29

Reply:
Ok, I took a second look at this and have still decided "no" for targets II. We _could_ potentially consider adding it into targets III when we talk about dynamic branching, but I think it would be better suited to a future targets 4 lesson. Or we could point them to the targets manual (when using dynamic branches, it boils down to [workspaces](https://books.ropensci.org/targets/debugging.html#workspaces))